src/auth/executableresponse.ts

```diff
@@ -0,0 +1,166 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const SAML_SUBJECT_TOKEN_TYPE = 'urn:ietf:params:oauth:token-type:saml2';
+const OIDC_SUBJECT_TOKEN_TYPE1 = 'urn:ietf:params:oauth:token-type:id_token';
+const OIDC_SUBJECT_TOKEN_TYPE2 = 'urn:ietf:params:oauth:token-type:jwt';
+
+/**
+ * Interface defining JSON format of response from 3rd party executable run by
+ * pluggable auth client.
+ */
+export interface ExecutableResponseJson {
+  /**
+   * Version of the response, only version 1 currently supported.
+   * Always required.
+   */
+  version: number;
+  /**
+   * Whether the executable ran successfully. Always required.
+   */
+  success: boolean;
+  /**
+   * Epoch time for expiration of the token in seconds, required for successful
+   * responses.
+   */
+  expiration_time?: number;
+  /**
+   * Type of subject token in the response, currently supported values are:
+   * urn:ietf:params:oauth:token-type:saml2
+   * urn:ietf:params:oauth:token-type:id_token
+   * urn:ietf:params:oauth:token-type:jwt
+   */
+  token_type?: string;
+  /**
+   * Error code from executable, required when unsuccessful.
+   */
+  code?: string;
+  /**
+   * Error message from executable, required when unsuccessful.
+   */
+  message?: string;
+  /**
+   * ID token to be used as a subject token when token_type is id_token or jwt.
+   */
+  id_token?: string;
+  /**
+   * Response to be used as a subject token when token_type is saml2.
+   */
+  saml_response?: string;
+}
+
+/**
+ * Class defining a response from a 3rd party executable run by the pluggable
```

Comment:
Defines the response of  a 3rd party executable run by the pluggable auth client.